src/Voron/Impl/Backup/FullBackup.cs

```diff
@@ -96,16 +96,20 @@ public class StorageEnvironmentInformation
             try
             {
                 long allocatedPages;
-                var writePesistentContext = new TransactionPersistentContext(true);
-                var readPesistentContext = new TransactionPersistentContext(true);
-                using (var txw = env.NewLowLevelTransaction(writePesistentContext, TransactionFlags.ReadWrite)) // so we can snapshot the headers safely
+                var writePersistentContext = new TransactionPersistentContext(true);
+                var readPersistentContext = new TransactionPersistentContext(true);
+                using (var txw = env.NewLowLevelTransaction(writePersistentContext, TransactionFlags.ReadWrite)) // so no new journal files will be created
+                using (env.Journal.Applicator.TakeFlushingLock()) // prevent from running JournalApplicator.UpdateDatabaseStateAfterSync() concurrently
```

Comment:
Do we need to worry about deadlocks?

Reply:
I don't think so. We get this lock for short duration.

Reply:
The actual problem is do we have anywhere where we take the flushing lock and then the write tx.

Reply:
You are right. Awesome catch. Working on fixing it.

Reply:
Also, maybe make sure that this is _always_ the case? So if we try to take tx write lock, we check that the flushing lock isn't already taken (debug only?).
That way, we can be sure that we are never getting them in the wrong order.

Reply:
> So if we try to take tx write lock, we check that the flushing lock isn't already taken (debug only?).

This is actually valid case - we can open a write tx while the flushing lock is taken. The assertion needs to be the other way around - when getting the flushing lock we need to ensure that we didn't take the write tx lock.

Please check it: https://github.com/ravendb/ravendb/pull/13787/commits/f555de1456ed0677cfd62f84a8fa43093b1f13f0